packages/opentelemetry-api/src/diag/logLevel.ts

```diff
@@ -0,0 +1,159 @@
+  ERROR = 0,
+
+  /** Identifies a warning scenario */
+  WARN = 1,
+
+  /** General informational log message */
+  INFO = 2,
+
+  /** General debug log message */
+  DEBUG = 3,
+
+  /**
+   * Detailed trace level logging should only be used for development, should only be set
+   * in a development environment.
+   */
+  VERBOSE = 4,
+
+  /** Used to set the logging level to include all logging */
+  ALL = 9999,
+}
+
+/**
+ * This is equivalent to:
+ * type LogLevelString = 'NONE' | TERMINAL' | 'CRITICAL' | 'ERROR' | 'WARN' | 'INFO' | 'DEBUG' | 'VERBOSE' | 'ALL';
+ */
+export type DiagLogLevelString = keyof typeof DiagLogLevel;
+
+/**
+ * Mapping from DiagLogger function name to Legacy Logger function used if
+ * the logger instance doesn't have the DiagLogger function
+ */
+const fallbackLoggerFuncMap: { [n: string]: keyof Logger } = {
+  terminal: 'error',
+  critical: 'error',
+  error: 'error',
+  warn: 'warn',
+  info: 'info',
+  debug: 'debug',
+  verbose: 'debug',
+  forcedInfo: 'info',
+};
+
+/** Mapping from DiagLogger function name to logging level. */
+const levelMap: { n: keyof DiagLogger; l: DiagLogLevel; f?: boolean }[] = [
+  { n: 'terminal', l: DiagLogLevel.TERMINAL },
+  { n: 'critical', l: DiagLogLevel.CRITICAL },
+  { n: 'error', l: DiagLogLevel.ERROR },
+  { n: 'warn', l: DiagLogLevel.WARN },
+  { n: 'info', l: DiagLogLevel.INFO },
+  { n: 'debug', l: DiagLogLevel.DEBUG },
+  { n: 'verbose', l: DiagLogLevel.VERBOSE },
+  { n: 'forcedInfo', l: DiagLogLevel.INFO, f: true },
+];
+
+/**
+ * Create a Diagnostic logger which limits the messages that are logged via the wrapped logger based on whether the
+ * message has a {@link DiagLogLevel} equal to the maximum logging level or lower, unless the {@link DiagLogLevel} is
+ * NONE which will return a noop logger instance. This can be useful to reduce the amount of logging used for the
+ * system or for a specific component based on any local configuration.
+ * If you don't supply a logger it will use the global api.diag as the destination which will use the
+ * current logger and any filtering it may have applied.
+ * To avoid / bypass any global level filtering you should pass the current logger returned via
+ * api.diag.getLogger() however, any changes to the logger used by api.diag won't be reflected for this case.
+ * @param maxLevel - The max level to log any logging of a lower
+ * @param logger - The specific logger to limit, if not defined or supplied will default to api.diag
+ * @implements {@link DiagLogger}
+ * @returns {DiagLogger}
+ */
+
+export function createLogLevelDiagLogger(
+  maxLevel: DiagLogLevel,
+  logger?: DiagLogger | null
+): DiagLogger {
+  if (maxLevel < DiagLogLevel.NONE) {
+    maxLevel = DiagLogLevel.NONE;
+  } else if (maxLevel > DiagLogLevel.ALL) {
+    maxLevel = DiagLogLevel.ALL;
+  }
+
+  if (maxLevel === DiagLogLevel.NONE) {
+    return createNoopDiagLogger();
+  }
+
+  if (!logger) {
+    logger = DiagAPI.instance().getLogger();
```

Comment:
This is inconsistent with the comments above, will fix and create additional unit test to validate.